programs/drift_vaults/src/state/math.rs

```diff
@@ -0,0 +1,112 @@
+use anchor_lang::prelude::*;
+use drift::math::safe_math::SafeMath;
+
+pub fn token_a_to_token_b(
+    token_a_amount: u64,
+    token_a_price: i64,
+    token_a_decimals: u32,
+    token_b_price: i64,
+    token_b_decimals: u32,
+) -> Result<u64> {
+    let quote_amount: u128 = (token_a_amount as u128).safe_mul(token_a_price as u128)?;
```

Comment:
do you preserve more precision if you do token_a_amount * 10^token_b_decimals*token_a_price/token_b_price/token_a_decimals?